### PR TITLE
Add compose environment

### PR DESCRIPTION
**Pull request by guilhermebr**
This PR makes possible run tsuru for development using docker-compose.

The compose file run tsuru dependences and create the tsuru api using the tsurud binary from local build folder.

Calling `build-compose.sh` will build tsurud inside a docker container, build the api image and run docker-compose up -d.

How-to: docs/contributing/compose.rst

### CODE REVIEW

File: docs/contributing/compose.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
+.. Copyright 2017 tsuru authors. All rights reserved.
+   Use of this source code is governed by a BSD-style
+   license that can be found in the LICENSE file.
+
++++++++++++++++++++++++++++++++++++++++++++++++
+Building a development environment with Docker Compose.
++++++++++++++++++++++++++++++++++++++++++++++++
```
Comment (andrestc) [inline docs/contributing/compose.rst]: I think the `+++++++++++++++++++++++++++++++++++++++++++++++` lines need to be the same length as `Building a development environment with Docker Compose.` else sphinx complains

**Comment by andrestc** [comment]
LGTM, just added a doc comment

**Comment by cezarsa** [comment]
Maybe set `chmod +x build-compose.sh`?

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        privileged: true
+        command: --insecure-registry registry:5000
+        ports:
+            - "2375:2375"
+            - "32768-32772:32768-32772"
+        networks:
+            tsuru_net:
+                ipv4_address: 172.42.0.20
+
+networks:
```
Comment (avelino) [inline docker-compose.yml]: Why use IP? Wouldn't it be better (and transparent for all operating systems) use hostname (that solves docker)?

Comment (guilhermebr) [inline docker-compose.yml]: @avelino I already removed IP from node1, but API needs because the agent-deploy can't access docker dns to resolve API name.

**Pull request merged**